app/controllers/dashboards_controller.rb

```diff
@@ -1,4 +1,30 @@
 # frozen_string_literal: true
 class DashboardsController < ApplicationController
-  def index; end
+  include Flashable
+
+  def index
+    @uploads = Upload.last_uploads
+  end
+
+  def build
+    @version = InstitutionBuilder.run(current_user)
```

Comment:
This rewrite should be as fast or faster than the current queries, and should not pose any burden. The problem with building is that we have a design that does not require any particular order in uploading. Otherwise we would have to require the user to upload new Weam and Crosswalk CSVs (if present) prior to any other CSV. That would be a different functionality.